flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,660 @@
+        """Create a Ombi Entry from an TVDB ID."""
+
+        headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+        headers.update(auth())
+
+        url = f"{base_url}/api/v2/Search/{cls.entry_type}/moviedb/{tmdb_id}"
+
+        response = requests.get(url, headers=headers)
+
+        if not response.ok:
+            log.error(f"{url} return {response.reason}")
+            # Do I throw an exception here? Is that allowed from a plugin?
+            return None
+
+        data = response.json()
+
+        # Their is a bug in OMBI where if you get a movie by its TMDB ID
+        # then the theMovieDbId field will be blank for some reason...
+        data['theMovieDbId'] = tmdb_id
+
+        return OmbiTv(base_url, auth, data)
+
+
+class OmbiSet(MutableSet):
+    """The schema for the Ombi managed list."""
+
+    supported_ids = ['imdb_id', 'tmdb_id', 'ombi_id']
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'api_key': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'type': {'type': 'string', 'enum': ['shows', 'seasons', 'episodes', 'movies']},
+            'status': {
+                'type': 'string',
+                'enum': ['approved', 'available', 'requested', 'denied'],
+                'default': 'requested',
+            },
+            'include_year': {'type': 'boolean', 'default': False},
+            'include_ep_title': {'type': 'boolean', 'default': False},
+        },
+        'oneOf': [{'required': ['username', 'password']}, {'required': ['api_key']}],
+        'required': ['url', 'type'],
+        'additionalProperties': False,
+    }
+
+    @property
+    def immutable(self):
+        return False
+
+    def __init__(self, config: Dict[str, Any]):
+        self.config: Config = config
+        self._items: Optional[List[Entry]] = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def add(self, entry: Entry):
+        log.info(f"Adding {entry['title']} to ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+        self.invalidate_cache()
+
+    def __ior__(self, entries: List[Entry]):
+        for entry in entries:
+            self.add(entry)
+
+    def discard(self, entry: Entry):
+        """Removes an entry from OMBI by marking it as available.
+
+        Args:
+            entry (Entry): An item from a task.
+        """
+
+        log.info(f"Removing {entry['title']} from ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
```

Comment:
Was looking at the issue of finding things and I have some questions as both add and remove call  get_ombi_entry(entry)

To get something get_ombi_entry(entry) calls from_id(request, entry) which doesn't do much other than  calls a more  specific from_tmdb_id(request, entry['tmdb_id']

get_ombi_entry(entry) already has the all of the entry data and IDs, why doesn't it just go in order of preference checking if there is a valid IT and built the search query?

Could just check if each exists and is not blank and search by the first one that is available, or try each in a search and if they fail try the next.

tmdbid
imdb_id
ombi_request_id (will be available if you are using an ombi list as an input to remove things from an ombi list for example)
title (the ombi API supports searching by title) but for this one you would have to maybe do a safety check that there was only one result in the search.

Reply:
I don't know what IDs we will be able to support. The Ombi API is very poor and basically only supports `tmdb_id` for movies and `tvdb_id` for TV. It doesn't even seem to support its own "ombi_id" for CRUD operations. I will have to look and see if `ombi_request_id` will work for most API calls. 

The other reason I did this, is to make unit testing easier. `get_ombi_entry` currently only has to decide if the `ombi_list` is configured for `movies` or not. If I have it also figuring out which ID the entry has, that will add a lot of permutations that need to be tested. 

I could probably refactor this even further and make it more abstract but I can't until we know the ID's we will support and it honestly doesn't make sense to spend more time on a working function when other parts of the plugin are not working correctly. 

We could spend a lot of time building the perfect `ombi_list` and then when we release it, we find out the majority of users expected it to work differently. Which is why my priority has always been getting a working minimal version out to users. So that we can expand on it using feedback from them.

Which means our first iteration might only support `tmdb_id` and `tvdb_id`.

Reply:
>The Ombi API is very poor and basically only supports tmdb_id for movies and tvdb_id for TV.

Title:Five Nights at Freddy's Request:1917 TMDB:507089 imdbId:tt4589218

These two return the same v1 info
/api/v1/Request/movie/search/Five%20Nights%20at%20Freddy%27s
/api/v1/Request/movie/info/1917

These two return the same v2 info
/api/v2/Search/movie/imdb/tt4589218'
/api/v2/Search/movie/507089

I think the above 4 calls should all find the same request and provide enough info to mark it available or delete it.

>The other reason I did this, is to make unit testing easier.

If it makes unit testing easier then by all means, just tracing the code it didn't steam like each method / function was doing enough to warrant it going 3 deep. The code for each of the lowest methods was nearly identical which indicated to me that it was redundant and should probably all be rolled up into get_ombi_entry(entry).

It could also be resolved by creating a from_request_ID  and from_title in theory if we want to keep the abstraction.

>We could spend a lot of time building the perfect ombi_list and then when we release it, we find out the majority of users expected it to work differently.

If an ombi list is the source, you would expect it would contain enough info to find the record and update it in another ombi list.

`  ombi_list_expire:
    disable: seen
    ombi_list:
      url: '{? ombi.url ?}'
      api_key: '{? ombi.api_key ?}'
      type: movies
    if:
      - ombi_request_id <= 1700 : accept
    list_remove:
      - ombi_list:
          url: '{? ombi.url ?}'
          api_key: '{? ombi.api_key ?}'
          type: movies`

Reply:
@bagobones @gazpachoking I have opened an issue https://github.com/Ombi-app/Ombi/issues/5041 about the Ombi `Search` and `Request` API's to see if they can make some changes that will make our work easier.

Reply:
I was wrong and I have `imdb_id` and `tmdb_id` working work movies. I still need to test tv but I'm resolving this conversion for now.